homeassistant/components/nintendo_parental/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Sensor platform for Nintendo Parental."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from enum import StrEnum
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import NintendoParentalConfigEntry, NintendoUpdateCoordinator
+from .entity import Device, NintendoDevice
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+class NintendoParentalSensor(StrEnum):
+    """Store keys for Nintendo Parental sensors."""
+
+    PLAYING_TIME = "playing_time"
+    TIME_REMAINING = "time_remaining"
+
+
+@dataclass(kw_only=True, frozen=True)
+class NintendoParentalSensorEntityDescription(SensorEntityDescription):
+    """Description for Nintendo Parental sensor entities."""
+
+    value_fn: Callable[[Device], int | float | None]
+
+
+SENSOR_DESCRIPTIONS: tuple[NintendoParentalSensorEntityDescription, ...] = (
+    NintendoParentalSensorEntityDescription(
+        key=NintendoParentalSensor.PLAYING_TIME,
+        translation_key=NintendoParentalSensor.PLAYING_TIME,
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.today_playing_time,
+    ),
+    NintendoParentalSensorEntityDescription(
+        key=NintendoParentalSensor.TIME_REMAINING,
+        translation_key=NintendoParentalSensor.TIME_REMAINING,
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.today_time_remaining,
+    ),
```

Comment:
we could add a suggested unit of measurement if that fits

Reply:
These are always in minutes when returned from the API itself. Not sure if there is a benefit to adding a suggested unit of measurement?

Reply:
I am wondering if most users consume this in minutes or hours